src/test/java/uk/gov/hmcts/befta/dse/ccd/definition/converter/EnvironmentURLUtilsTest.java

```diff
@@ -0,0 +1,243 @@
+            "  \"CallBackURLSubmittedEvent\" : \"%s%s\",\n" +
+            "  \"RetriesTimeoutURLSubmittedEvent\" : \"\",\n" +
+            "  \"SecurityClassification\" : \"Public\"\n" +
+            "}, {\n" +
+            "  \"CaseTypeID\" : \"FT_NoCAutoApprovalCaseType\",\n" +
+            "  \"ID\" : \"ApplyNoCDecision\",\n" +
+            "  \"Name\" : \"Apply NoC Decision\",\n" +
+            "  \"Description\" : \"Apply Notice of Change Request\",\n" +
+            "  \"CallBackURLAboutToStartEvent\" : \"%s%s\",\n" +
+            "  \"RetriesTimeoutAboutToStartEvent\" : \"\",\n" +
+            "  \"CallBackURLAboutToSubmitEvent\" : \"\",\n" +
+            "  \"RetriesTimeoutURLAboutToSubmitEvent\" : \"\",\n" +
+            "  \"CallBackURLSubmittedEvent\" : \"\",\n" +
+            "  \"RetriesTimeoutURLSubmittedEvent\" : \"\",\n" +
+            "  \"SecurityClassification\" : \"Public\"\n" +
+            "} ]";
+
+    private static final String NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_DEFAULT_HOST = "http://localhost:4454";
+    private static final String NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_HOST =
+            String.format("${MCA_API_BASE_URL:%s}", NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_DEFAULT_HOST);
+    private static final String NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_PATH = "/noc/check-noc-approval";
+
+    private static final String APPLY_NOC_DECISION_SUBMIT_EVENT_DEFAULT_HOST = "http://localhost:5555";
+    private static final String APPLY_NOC_DECISION_SUBMIT_EVENT_HOST =
+            String.format("${TEST_STUB_SERVICE_BASE_URL:%s}", APPLY_NOC_DECISION_SUBMIT_EVENT_DEFAULT_HOST);
+    private static final String APPLY_NOC_DECISION_SUBMIT_EVENT_PATH =
+            "/callback_returning_simulated_completed_noc_request";
+
+    private static final String CREATE_CASE_CALLBACK_MID_EVENT_DEFAULT_HOST
+            = "http://ccd-test-stubs-service:5555";
+    private static final String CREATE_CASE_CALLBACK_MID_EVENT_HOST
+            = String.format("${TEST_STUB_SERVICE_BASE_URL:%s}", CREATE_CASE_CALLBACK_MID_EVENT_DEFAULT_HOST);
+    private static final String CREATE_CASE_CALLBACK_MID_EVENT_PATH
+            = "/case_type/fe-functional-test/mid_event_dynamic_list";
+
+    @BeforeEach
+    void setup() throws IOException {
+        caseEventJson = objectMapper.readTree(
+                String.format(CASE_EVENT_CONTENT_TEMPLATE,
+                        NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_HOST, NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_PATH,
+                        APPLY_NOC_DECISION_SUBMIT_EVENT_HOST, APPLY_NOC_DECISION_SUBMIT_EVENT_PATH)
+        );
+    }
+
+    @Test
+    void unsupportedSheetNamesShouldNotModifyURLs() throws JsonProcessingException, MalformedURLException {
+        JsonNode nullJsonNode = objectMapper.nullNode();
+
+        JsonNode modifiedJsonNode = EnvironmentURLUtils.updateCallBackURLs(nullJsonNode, "AFileNameThatIsNotCaseEvent");
+        assertEquals(nullJsonNode, modifiedJsonNode);
```

Comment:
I like your other inline comments 👍, they are not as cluttered as some standard assert messages can be (although these help in failure reports).

Can we add a similar comment here: `// check JSON has mot been modified`

Reply:
but with the correct spelling: `// check JSON has not been modified`

Reply:
Resovled in https://github.com/hmcts/befta-fw/pull/123/commits/f7fbff8c44e377ce7f637f1419a516c10c184c61 and typo fixed in https://github.com/hmcts/befta-fw/pull/123/commits/633aeed3ecda977497ea1a0bf4bb34a3e41a012f